src/util/renderers/abstract.renderer.ts

```diff
@@ -0,0 +1,3 @@
+export abstract class Renderer {
+  abstract render(renderOpts: any): Promise<any>
```

Comment:
We should have a strict interface `renderOpts: any` this is really too error prone.

Reply:
Fixed